docs/MetricsSpecification.md

```diff
@@ -0,0 +1,164 @@
+# ADO.NET Metrics Specification
+
+## Introduction
+
+This specification describes how to add metrics (following the
+[OpenTelemetry Semantic Conventions for Database Metrics](https://github.com/open-telemetry/semantic-conventions/blob/main/specification/metrics/semantic_conventions/database-metrics.md))
+to ADO.NET providers using [System.Diagnostics.Metrics](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.metrics).
+
+## Background
+
+[System.Diagnostics.Metrics APIs](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/compare-metric-apis#systemdiagnosticsmetrics) are the newest cross-platform APIs for collecting performance metrics,
+and were designed in close collaboration with the OpenTelemetry project.
+OpenTelemetry semantic conventions are standards for generating consistent,
+accessible telemetry across different databases and client libraries.
+
+## General
+
+This specification should be read in conjunction with the [OpenTelemetry Semantic Conventions](https://github.com/open-telemetry/semantic-conventions/blob/main/specification/metrics/semantic_conventions/database-metrics.md).
+The OpenTelemetry Semantic Conventions MUST be followed except where this specification explicitly overrides them.
+
+ADO.NET Providers MUST use the [`System.Diagnostics.Metrics.Meter` class](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.metrics.meter) to expose metrics.
+It MUST be created with the name of the library and its version:
+
+```csharp
+public static Meter Meter { get; } = new("Xyzsql", "1.0.0");
+```
+
+## Pool Names
+
+OpenTelemetry Semantic Conventions assume that connection pools can be named.
```

Comment:
Yeah - though note that no such concept currently exists in any ADO.NET provider as far as I'm aware. We did implement support for per-pool metrics in Npgsql via the older event counters API, using the connection string as the name - but that's not great.

I think introducing a name concept at the data source level is a great idea, opened https://github.com/npgsql/npgsql/issues/5108 to track on the Npgsql side.

API nit: I'd call it `HasName` rather than `UseName`.